packages/siwe/lib/client.ts

```diff
@@ -297,27 +297,35 @@ export class SiweMessage {
 			} finally {
 				/** Match signature with message's address */
 				if (addr !== this.address) {
-					let isValid = false;
-					try {
-						/** Try resolving EIP-1271 if address doesn't match */
-						isValid = await checkContractWalletSignature(
-							this,
-							signature,
-							opts.provider
-						);
-					} catch (_) {
-						isValid = false;
-					} finally {
-						if (!isValid) {
-							assert({
-								success: false,
-								data: this,
-								error: new SiweError(
-									SiweErrorType.INVALID_SIGNATURE,
-									addr,
-									`Resolved address to be ${this.address}`
-								),
-							});
+					/** Checks and call for failureCallback before trying EIP-1271 resolution */
+					if (opts.failureCallback) {
+							const result = await opts.failureCallback(params, opts, this);
+							if(!result.success) {
+								assert(result)
+							} 
+					} else {
+						let isValid = false;
```

Comment:
Should the existence of a failureCallback mean we don't try EIP-1271 validation? I would suggest that the failure callback code should go in the finally block of the below try-catch.